tests/TestCase/Filesystem/FileTest.php

```diff
@@ -123,6 +123,69 @@ function_exists('mime_content_type') &&
         $this->assertInstanceOf('Cake\Filesystem\Folder', $result);
     }
 
+    /**
+     * testUtf8Filenames
+     *
+     * @link https://github.com/cakephp/cakephp/issues/11749
+     * @return void
+     */
+    public function testUtf8Filenames()
+    {
+        $File = new File(TMP . 'tests/permissions/نام فارسی.php', true);
+        $this->assertEquals('نام فارسی', $File->name());
+        $this->assertTrue($File->exists());
+        $this->assertTrue($File->readable());
+    }
+
+    /**
+     * Test _basename method
+     * @dataProvider baseNameValueProvider
+     * @return void
+     */
+    public function testBasename($path, $suffix)
+    {
+        $class = new \ReflectionClass('Cake\Filesystem\File');
+        $method = $class->getMethod('_basename');
+        $method->setAccessible(true);
+        if ($suffix === null) {
+            $this->assertEquals(basename($path), $method->invokeArgs(null, [$path]));
+        } else {
+            $this->assertEquals(basename($path, $suffix), $method->invokeArgs(null, [$path, $suffix]));
```

Comment:
If you want to call this method why is it not public? Coupling tests to 'implementation details' makes tests harder to maintain as private features are now public.

Reply:
we can add/move these tests to other public tests
i just want to know `_basename()` method is equal to `basename()` 
And i no want to have public method at this time

Reply:
Can we use a public interface of the class then?

Reply:
>Can we use a public interface of the class then?

1)having extended class withing testCase and having a method for wrapping _basename() 
like the https://github.com/cakephp/cakephp/blob/master/tests/TestCase/Mailer/EmailTest.php#L89-L92 in to testcase?

2) or moving test to public methods?

in case 2 , not fully - we can not cover all case - i'm not sure

Reply:
Can we not create empty files for each case and check their names?

Reply:
I'll move these tests  to `name()` without having real file

Reply:
Could the behavior of `_basename` be tested based on how file creation and `name()` are working?